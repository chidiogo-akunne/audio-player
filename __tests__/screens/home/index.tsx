import HomeScreen from '@screens/home';
import renderer from 'react-test-renderer';
import ThemeProvider from '@theme';
import { ContextThemeProvider } from '@context/index';
import { QueryClient, QueryClientProvider } from 'react-query';

jest.useFakeTimers();

// Create a react queultry client
const queryClient = new QueryClient();

describe('Home', () => {
  const tree = renderer.create(
    <ThemeProvider>
      <QueryClientProvider client={queryClient}>
        <ContextThemeProvider>
          <HomeScreen />
        </ContextThemeProvider>
      </QueryClientProvider>
    </ThemeProvider>
  );
  it(`renders correctly`, async () => {
    await expect(tree).toMatchSnapshot();
  });
});
