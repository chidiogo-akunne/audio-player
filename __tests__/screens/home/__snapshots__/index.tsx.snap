// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F8F8F8",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 39,
          "width": "100%",
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={[Function]}
      contentContainerStyle={
        Object {
          "paddingBottom": 196,
          "paddingHorizontal": 29,
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={true}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginTop": 20,
          "width": "100%",
        }
      }
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          accessibilityRole="progressbar"
          accessibilityState={
            Object {
              "busy": true,
            }
          }
          accessible={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "marginBottom": "auto",
                "marginTop": "auto",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 78,
                "opacity": 1,
                "width": 78,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 78,
                    "transform": Array [
                      Object {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 78,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 39,
                      "overflow": "hidden",
                      "width": 78,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 78,
                        "transform": Array [
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "rotate": "-165deg",
                          },
                        ],
                        "width": 78,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 39,
                          "overflow": "hidden",
                          "width": 78,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "borderColor": "#000000",
                            "borderRadius": 39,
                            "borderWidth": 7.8,
                            "height": 78,
                            "width": 78,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 78,
                    "transform": Array [
                      Object {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 78,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 39,
                      "overflow": "hidden",
                      "top": 39,
                      "width": 78,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 78,
                        "transform": Array [
                          Object {
                            "translateY": -39,
                          },
                          Object {
                            "rotate": "345deg",
                          },
                        ],
                        "width": 78,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 39,
                          "overflow": "hidden",
                          "width": 78,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "borderColor": "#000000",
                            "borderRadius": 39,
                            "borderWidth": 7.8,
                            "height": 78,
                            "width": 78,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
